### DISCUSSION TITLE
How to create plugin for realativley symple WSS:// Urls?

### DISCUSSION THREAD

**Discussion by AdmnJ** [streamlink/streamlink#5079] [General]
I have read a few plugins that contain WSS:// in them but I am having a hard time understanding how one implements their own plugin for one of these sites. Trying to make a plugin for https://vrcdn.live which generates links following this convention typically. 
`https://vrcdn.live/preview/USERNAME -> video-url="wss://stream.vrcdn.live/live/USERNAME.flv"` The site is often used by DJs for VR clubs and most of the audience just listens through the in world video players... but when the game lags the video players sometimes suffer as well. One of the ways around this is to load the stream in browser but considering I am here I am sure you can tell what method I would prefer.

**Comment by bastimeyer** [comment]
WebSocket streams require custom stream implementations, and this depends on whether binary stream data is served via the websocket, or whether the websocket connection implements a specific streaming protocol, or whether addresses of standard streaming protocols are sent on the websocket connection.

Examples:
- UStreamTV (DASH-like protocol via websocket)
  https://github.com/streamlink/streamlink/blob/ae7583473626a09f870fbdb71e4a988a83664ea6/src/streamlink/plugins/ustreamtv.py#L66
- NicoLive (HLS stream URL read from websocket)
  https://github.com/streamlink/streamlink/blob/ae7583473626a09f870fbdb71e4a988a83664ea6/src/streamlink/plugins/nicolive.py#L25
- Twitcasting (binary stream data served via websocket)
  https://github.com/streamlink/streamlink/blob/ae7583473626a09f870fbdb71e4a988a83664ea6/src/streamlink/plugins/twitcasting.py#L94

**Comment by aaronfrancis635** [comment]
> I have read a few plugins that contain WSS:// in them but I am having a hard time understanding how one implements their own plugin for one of these sites. Trying to make a plugin for https://vrcdn.live which generates links following this convention typically. `https://vrcdn.live/preview/USERNAME -> video-url="wss://stream.vrcdn.live/live/USERNAME.flv"` The site is often used by DJs for VR clubs and most of the audience just listens through the in world video players... but when the game lags the video players sometimes suffer as well. One of the ways around this is to load the stream in browser but considering I am here I am sure you can tell what method I would prefer.

Hi AdmnJ, VRCDN has the ability to be loaded in VLC etc using the RTSP/RTMP URLs natively, if you need the WSS://, it sends binary FLV data.

👍
